src/admin.rs

```diff
@@ -350,6 +362,139 @@ where
     custom_protocol_response_ok(stream, "SET").await
 }
 
+/// Bans a host from being used
+async fn ban<T>(stream: &mut T, tokens: Vec<&str>) -> Result<(), Error>
+where
+    T: tokio::io::AsyncWrite + std::marker::Unpin,
+{
+    let host = match tokens.get(1) {
+        Some(host) => host,
+        None => return error_response(stream, "BAN command requires a hostname to ban").await,
```

Comment:
non blocking, can be done in a follow up: do we want to accept a duration string for how long to ban it for?